### DISCUSSION TITLE
How to send_transaction to solidity smart contract using WalletConnectV2 and iOS?

### DISCUSSION THREAD

**Discussion by Ivan-Dosev** [WalletConnect/walletconnect-monorepo#4081] [Developer Support]
I have the following code:

    import SwiftUI
    import WalletConnectV2SwiftUI
    import WalletConnectModal
    import WalletConnectSwift
    import WalletConnectSign
    import UIKit
    import Combine

    struct AccountDetails {
    let chain: String
    let methods: [String]
    let account: String
    }

    struct ContentView: View {
    
    
    @State var account : String = ""
    
    let metadata = AppMetadata(name: "Swift Dapp",
                               description: "WalletConnect to App",
                               url:"www.walletconnect.com",
                               icons: [""])
    
    @StateObject var walletConnect = WalletConnectView(
        projectId: "2cd853f680bd02428bb89b1d2943f44f",
        name: "Swift Dapp",
        description: "WalletConnect to App",
        url: "www.walletconnect.com",
        icons: [""],
        supportedChainIds: [
            "eip155": ProposalNamespace(
                chains: [
                    Blockchain("eip155:137")!
                ],
                methods: [
                    "eth_sendTransaction"
                ], events: []
            )])
    
    
    var body: some View {
        
        
        VStack {
            Text("account: \(account)")
                .fontWeight(.thin)
            
            Spacer()
            
            
            if !account.isEmpty {
                Button(action: {
                    
                    
                    // TODO  send Transaction
                    
                    
                }, label: {
                    
                    Text("send Transaction")
                        .padding()
                        .background(RoundedRectangle(cornerRadius: 20).fill(Color.yellow.opacity(0.3)))
                })
            }else{
                
                
                Button {
                    walletConnect.connectWithWallet()
                } label: {
                    Text("Connect To Wallet")
                        .padding()
                        .background(RoundedRectangle(cornerRadius: 20).fill(Color.green.opacity(0.3)))
                }
                .padding(.vertical, 30)
            }
            
        }
        .padding()
        .onChange(of: walletConnect.account, perform: { accounts in
            
            self.account = accounts.first?.address ?? "..."
            
            print("Wallet Address: ", accounts.first?.address ?? "")
        })
        .onChange(of: walletConnect.rejectedReason, perform: { reason in
            print("Wallet Connection Error", reason)
        })
    }
    
    }

It is a very simple code where users can connect to Metamask or any other blockchain wallet using WalletConnectV2. My question now is, once I am connected how can I send transaction and interact with a smart contract. Lets say I have a smart contract address, ABI and a function I want to call. How can I integrate it inside my app? In the previous version V1, there was a demo but in V2 I can not find anything. Inside the official WalletConnect documentation there is only one example how to connect but I also want call some smart contract function after that.

**Comment by benchegimah07** [comment]
Certainly, as an online entrepreneur and Rural Poultry Farmers Credit Union Agent. The Need To Source Affiliates Organizations Like YOU To Enable Me In Supporting The Poor And The Needy During The Farming Seasons All Year Round. With You, Together We Shall Build Up The Broken Walls In Our Communities And The World At Large.
CHEERIO!!!
![IMG_20231002_100833](https://github.com/WalletConnect/walletconnect-monorepo/assets/152743255/0d468214-32d3-4a88-a2a0-23ab38e7a473)
